spec/avram/queryable_spec.cr

```diff
@@ -856,6 +856,97 @@ describe Avram::Queryable do
     end
   end
 
+  describe "#group_count" do
+    context "when no records exist" do
+      it "0 grouped columns returns { [] => 0 }" do
+        query = UserQuery.new
+
+        query.group_count.should eq({ [] of String => 0 })
```

Comment:
I'm not so sure about this... To me, this seems like a mistake. You're saying "Give me a grouped count on a non-grouped query". I get that canonically it's correct, but there's a few things we do in Lucky where we say "even though it's semantically correct, it's still more likely to cause issues". For example with empty strings. We treat those the same as `nil` even though they're not.

@matthewmcgarvey what do you think about this raising an error instead? If your query isn't grouped, should you still be allowed to call this method?

Reply:
I'm fine with that. Just checked and Rails does not allow you to call group without a column https://github.com/rails/rails/blob/9a0d951f3849ac84a2a9baa9e8080ede4c912183/activerecord/lib/active_record/relation/query_methods.rb#L352-L355

Reply:
Rails doesn't allow you to call a group with no columns, and neither do we -- this is only allowing you to return the same type without requiring a group

Reply:
I still think we need to make it an error, but I might have a way to make it a compile-time error... We can roll with what you have now, and I'll do a PR a bit later to try it out. Thanks for the explanation, and all the work on this!

Basically I want to try setting a nilable bit once you've added a group, then do a check on the args like [rails does](https://github.com/rails/rails/blob/9a0d951f3849ac84a2a9baa9e8080ede4c912183/activerecord/lib/active_record/relation/query_methods.rb#L1677-L1678) but it'll check it at compile-time forcing you to call `group()` at least once before calling this method. 😄

Reply:
If it's compile time I feel that's a lot more tolerable than what I initially had which I believe is a runtime exception